js/ad_manager_vast.js

```diff
@@ -1616,6 +1645,218 @@ OO.Ads.manager(function(_, $) {
+                  if (!this.testMode){
+                    this.ajax(adSource.adTagURI, this.onVastError, 'xml', adObject);
+                  }
+                }
+              }
+              else {
+                OO.log("VAST, VMAP: Error creating Ad Object");
+              }
+            }
+          }
+        }
+      }, this);
+    };
+
+    /**
+     * Helper function to find all node names with "vmap:TrackingEvents" / "TrackingEvents", and pick only
+     * the elements with "vmap:TrackingEvents".
+     * Note: must search for both "vmap:TrackingEvents" and "TrackingEvents" because of weird issue where
+     * Chrome cannot find "vmap:TrackingEvents" unless another selector is specified.
+     * @private
+     * @method Vast#_findVMAPTrackingEvents
+     * @param {object} adBreakElement The adBreak element to search
+     * @returns {object[]} The filtered array with only vmap tracking events.
+     */
+    var _findVMAPTrackingEvents = _.bind(function(adBreakElement) {
+      var trackingEventsElement = $(adBreakElement).find("vmap\\:TrackingEvents, TrackingEvents");
+      var VMAPTrackingEventsElement = _.filter(trackingEventsElement.toArray(), function(trackingEventElement) {
+        return (trackingEventElement.tagName.toLowerCase().indexOf("vmap:") > -1);
+      }, this);
+      return VMAPTrackingEventsElement;
+    }, this);
+
+    /**
+     * Helper function to convert VMAP tracking events into objects with attributes as properties.
+     * @private
+     * @method Vast#_parseVMAPTrackingEvents
+     * @param {object} trackingEventsElement The tracking events element
+     * @returns {object[]} The array of tracking event objects.
+     */
+    var _parseVMAPTrackingEvents = _.bind(function(trackingEventsElement) {
+      var trackingEvents = [];
+      var trackingElements = $(trackingEventsElement).find("vmap\\:Tracking, Tracking");
+      if (trackingElements.length > 0) {
+        _.each(trackingElements, function(trackingElement) {
+          var tracking = {};
+          trackingEvents.push(tracking);
+          tracking.url = $(trackingElement).text();
+          tracking.eventName = $(trackingElement).attr("event");
+        }, this);
+      }
+      return trackingEvents;
+    }, this);
+
+    /**
+     * Convert the adBreak attributes into an ad object that will be passed into _onVastResponse().
+     * @private
+     * @method Vast#_convertToAdObject
+     * @param {object} adBreak The adBreak object
+     * @returns {object} null if the timeOffset attribute does not match any format. Otherwise, the
+     * ad object is returned.
+     */
+    var _convertToAdObject = _.bind(function(adBreak) {
+      var adObject = {
+        /*
+         *ad_set_code: "",
+         *click_url: "",
+         *expires: 0,
+         *first_shown: 0,
+         *frequency: 2,
+         *public_id: "",
+         *signature: "",
+         *tracking_url: [],
+         *type: "",
+         *url: ""
```

Comment:
Are the rest of these properties important? I only use the "time" and "position_type" properties.